lib/KSelect/index.vue

```diff
@@ -60,123 +167,606 @@
+
+        const character = event.key.toString();
+        this.quickMatchString += character;
+        let matchingItems = this.annotatedOptions.filter(
+          option => option.startsWith && !option.disabled
+        );
+        if (matchingItems.length !== 0) {
+          matchingItems = sortby(matchingItems, [this.keys.label]);
+          matchingItems = sortby(matchingItems, item => item[this.keys.label].length);
+          this.highlightOption(this.options[matchingItems[0].index]);
+        }
+      },
+
+      // Highlights the previous valid option
+      highlightPreviousOption() {
+        const options = this.annotatedOptions;
+        let validPreviousOptionIndex = -1;
+        for (let i = 0; i < options.length; i++) {
+          if (options[i].validPreviousOption) {
+            validPreviousOptionIndex = i;
+          }
+        }
+        if (validPreviousOptionIndex !== -1) {
+          this.highlightOption(this.options[validPreviousOptionIndex]);
+        }
+      },
+
+      // Highlights the next valid option
+      highlightNextOption() {
+        const options = this.annotatedOptions;
+        const validNextOptionIndex = options.findIndex(option => option.validNextOption);
+        if (validNextOptionIndex !== -1) {
+          this.highlightOption(this.options[validNextOptionIndex]);
+        }
+      },
+
+      // Highlights the option
+      highlightOption(option, options = { autoScroll: true }) {
+        if (
+          !option ||
+          option.disabled ||
+          looseEqual(this.highlightedOption, option) ||
+          this.$refs.options.length === 0
+        ) {
+          return;
+        }
+
+        this.highlightedOption = option;
+        this.openDropdown();
+
+        if (options.autoScroll) {
+          const index = this.filteredOptions.findIndex(option =>
+            looseEqual(this.highlightedOption, option)
+          );
+          const optionToScrollTo = this.$refs.options[index];
+          if (optionToScrollTo) {
+            this.scrollOptionIntoView(optionToScrollTo.$el);
+          }
+        }
+      },
+
+      selectHighlighted() {
+        if (
+          this.highlightedOption &&
+          !this.highlightedOption.disabled &&
+          this.$refs.options.length > 0
+        ) {
+          this.selectOption(this.highlightedOption);
+        }
+      },
+
+      selectOption(option, options = { autoClose: true }) {
+        if (!option || option.disabled) {
+          return;
+        }
+
+        this.setValue(option);
+
+        this.$emit('select', option, {
+          selected: !this.isOptionSelected(option),
+        });
+
+        this.clearQuery();
+
+        if (options.autoClose) {
+          this.closeDropdown();
+        }
+      },
+
+      // Checks if option is highlighted
+      isOptionHighlighted(option) {
+        return looseEqual(this.highlightedOption, option);
+      },
+
+      isOptionSelected(option) {
+        return looseEqual(this.selection, option);
```

Comment:
Logic reduced warning!

Reply:
The same reason of `multiple` prop is here.